packages/key4hep-stack/package.py

```diff
@@ -9,14 +9,26 @@ class Key4hepStack(BundlePackage):
     
     version('0.1')
     version('nightly')
+
+    variant('cxxstd',
+            default='17',
+            values=('11', '14', '17'),
+            multi=False,
+            description='Use the specified C++ standard when building.')
     
     depends_on('podio@master', when="@nightly")
     depends_on('edm4hep@master', when="@nightly")
     depends_on('K4FWCore@master', when="@nightly")
     depends_on('guinea-pig@master', when="@nightly")
-    depends_on('dd4hep@master', when='@nightly')
     
     depends_on("edm4hep@0.1.0", when="@0.1")
     depends_on("K4FWCore@0.1.0", when="@0.1")
     depends_on('guinea-pig@1.2.2rc', when="@0.1")
+
+    depends_on('lcio')
+    depends_on('lcgeo')
+
+    # explicitly set cxxstd to avoid concretizer error
+    depends_on('root cxxstd={0}' % self.spec.variants['cxxstd'].value )
```

Comment:
Hi @vvolkl Unfortunately this doesn't work, because `self` is not defined here. I have tried a similar thing once before, but I haven't found a way to access the values of the variants in the current spec.

Reply:
Good catch, seems like I copied this from the wrong place. Here I'd have to use something similar to the code in the podio recipe. But maybe it's better to just set it explicitly to 17 like you suggest below.